aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -526,6 +529,188 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
 }
 
 
+template<int size0>
+void vectorized_cpu_hflip_channels_last_uint8(int64_t & i, char * C10_RESTRICT *data, const int64_t size, const int64_t stride) {
+
+#ifdef CPU_CAPABILITY_AVX2
+
+  constexpr auto vec_size = 256 / (8 * sizeof(uint8_t));
+
+  if (size > vec_size) {
+
+    const auto proc_vec_size = 2 * ((vec_size / 2) / size0);
+    const auto delta = vec_size - proc_vec_size * size0;
+    __m256i data_vec, reversed_vec;
+
+    // Example for N=3
+    // Data: (1 2 3) (4 5 6) (7 8 9) (10 11 12) (13 14 15) (16 17 18) (19 20 21) (22 23 24) (25 26 27) (28 29 30) (31 32 33)
+    // load by 2 parts
+    // R = [ (1 2 3) (4 5 6) (7 8 9) (10 11 12) (13 14 15) (16 | (16 17 18) (19 20 21) (22 23 24) (25 26 27) (28 29 30) (31 ]
+    // flip(R) ->
+    // R = [ 31 (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) | 16 (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3) ]
+    // Write in 2 parts
+    // Output pointer:                                                                                                       v
+    //                (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X) (X X X) (X X X)
+
+    // Output part 1: (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X 16)   (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+    // Output part 2: (X X 31) (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+
+    // Shuffle mask for N channels, N in [2, 8]
+    char mdata[16] = {};
+    for (int k=0; k<16; k++) {
+      int j = k / size0 + 1;
+      int v = (j * size0 - 1) - (k % size0);
+      v = std::min(v, (int) 15);
+      mdata[k] = v;
+    }
+    __m256i mask = _mm256_set_epi8(
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15], // first 128-bit lane
+
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15] // second 128-bit lane
+    );
+
+    // shift output pointer by size0
+    data[0] += size0 * stride;
+
+    for (; i < size - vec_size; i += proc_vec_size * size0) {
+
+      // load 256-bits by two 128-bits parts
+      auto a0 = _mm_loadu_si128((__m128i *) data[1]);
+      auto b0 = _mm256_castsi128_si256(a0);
+      auto a1 = _mm_loadu_si128((__m128i *) (data[1] + proc_vec_size / 2 * size0 * sizeof(uint8_t)));
```

Comment:
You should pre-compute this
```suggestion
      auto a1 = _mm_loadu_si128((__m128i *) (data[1] + proc_vec_half_stride));
```